utilities/ImpactUtilities.groovy

```diff
@@ -169,6 +169,16 @@ def createImpactBuildList() {
 		}
 
 	}
+	//include statically called sub programs when the main program changes
+     if (props.addSubmodulesToBuildList && props.addSubmodulesToBuildList.toBoolean()) {
```

Comment:
Either you need to introduce another property to enable the feature, or you need to loop through the intermediate build list to understand if the additional analysis is required via the property `addSubmodulesToBuildList` for a set of files.

Reply:
Dennis - I have made the changes to include perform the analysis for a set of files as well introduced logic to reduce the number of metadatastore hits. I have also moved the logic before the 'changed configuration' check.